transform/all_collection/spark/utils/common_df_transformations.py

```diff
@@ -0,0 +1,246 @@
+) -> None:
+    """Create boolean value whether record is open access or not, based on col_name"""
+    best_access_right = df.select(col_name).collect()
+    open_access_column = []
+
+    for access in best_access_right:
+        if access[col_name] in {
+            "OPEN",
+            "open access",
+            "fully_open_access",
+            "open_access",
+        }:
+            open_access_column.append(True)
+        else:
+            open_access_column.append(False)
+
+    harvested_properties[OPEN_ACCESS] = open_access_column
+
+
+def harvest_funder(df: DataFrame, harvested_properties: Dict) -> None:
+    """Harvest funder -> name and fundingStream as arr(arr(arr(fundingStream, <value>), arr(name, <value>>)))"""
+    projects_list = df.select("projects").collect()
+    funder_column = []
+
+    for projects in projects_list:
+        if projects["projects"]:
+            funder_list = []
+            for project in projects["projects"]:
+                funder = [
+                    ["fundingStream", project["funder"]["fundingStream"]],
+                    ["name", project["funder"]["name"]],
+                ]
+                funder_list.append(funder)
+            funder_column.append(funder_list)
+        else:
+            funder_column.append(None)
+
+    harvested_properties[FUNDER] = funder_column
+
+
+def harvest_url_and_document_type(df: DataFrame, harvested_properties: Dict) -> None:
+    """
+    Harvest url from instance.element.url as array(str)
+
+    Assumption:
+    - url has to be unique for specific record and it has to be a link
+    """
+    instances_list = df.select("instance").collect()
+    url_column = []
+    document_type_column = []
+
+    for instances in instances_list:
+        if instances["instance"]:
+            url_list = []
+            document_type_list = []
+            for instance in instances["instance"]:
+                if instance["url"]:
+                    for url in instance["url"]:
+                        if url and url not in url_list:
+                            url_list.append(url)
+
+                document_type_list.append(instance["type"])
+
+            url_column.append(url_list)
+            document_type_column.append(document_type_list)
+        else:
+            url_column.append(None)
+
+    harvested_properties[URL] = url_column
+    harvested_properties[DOCUMENT_TYPE] = document_type_column
+
+
+def harvest_country(df: DataFrame, harvested_properties: Dict) -> None:
+    """Harvest country from country.element.code as array(str)"""
+    countries_list = df.select("country").collect()
+    country_column = []
+
+    for countries in countries_list:
+        if countries["country"]:
+            country_list = []
+            for country in countries["country"]:
+                country_list.append(country["code"])
+
+            country_column.append(country_list)
+        else:
+            country_column.append(None)
```

Comment:
This could be also simplified

```
for countries in countries_list:
  if countries['country']:
    country_column.append([country['code'] for country in countries['country'])
    continue
  
  country_column,append(None)

Reply:
:clap: I have changed `harvest_research_community` also in that way.